### PR TITLE
refactor(RHINENG-6162): Refactor pdf generator

### PR DESCRIPTION
**Pull request by adonispuente**
In an attempt to diagnose the pdf generator issue ,this PR refactors bits of the Download.js file : https://issues.redhat.com/browse/RHINENG-6162

### CODE REVIEW

File: src/PresentationalComponents/ExecutiveReport/Download.js [hunk 1]
```diff
@@ -23,30 +24,29 @@ const DownloadExecReport = ({ isDisabled }) => {
...
+      ).map(({ data }) => data);
 
-      const report = BuildExecReport({
-        statsReports,
-        statsSystems,
-        topActiveRec,
-        intl,
-      });
+      const report = (
+        <BuildExecReport
```
Comment (mkholjuraev) [inline src/PresentationalComponents/ExecutiveReport/Download.js:43]: I believe that the report should not be pre-rendered when it is passed to the generator service. For example, [vulnerability](https://github.com/RedHatInsights/vulnerability-ui/blob/5e9c2cce07773dc95fea302bcd8b94b9a08233ea/src/Components/SmartComponents/Reports/DownloadSystemsReport.js#L85) also does not return pre-rendered component.

Comment (bastilian) [reply src/PresentationalComponents/ExecutiveReport/Download.js:43]: What do you mean with "pre-rendered"?

Comment (bastilian) [reply src/PresentationalComponents/ExecutiveReport/Download.js:43]: If you mean call the `BuildExecReport` component (properly) as a component then, no it should not be called as a function. Yes, it works and Vulnerability is doing it too this way, but it is not correct and can interfere with Reacts correct behaviour. 

If it returns JSX it is a component, and components should be called as such.

Comment (mkholjuraev) [reply src/PresentationalComponents/ExecutiveReport/Download.js:43]: ah, I got confused as I was looking at vulnerability while reviewing

**Review by bastilian — APPROVED** [review]
Lets get this merged! Thank you @adonispuente!

**Comment by bastilian** [comment]
@adonispuente For now we can ignore the coverage warnings, I think, but we should eventually properly extract things into hooks and have tests for those. In case this does not help with the download we should look into that next as the tests could also give a hint at what goes wrong.

**Pull request merged**